crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -230,8 +231,17 @@ where
         input: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
         let mut messages = Vec::new();
-
-        let tedge_event = ThinEdgeEvent::try_from(&input.topic.name, input.payload_str()?)?;
+        let mqtt_payload = input.payload_str()?;
+        let mqtt_topic = input.topic.name.clone();
```

Comment:
Minor. I would avoid to clone the topic and to extract the payload twice even when there is no error.